wgpu-hal/src/noop/buffer.rs

```diff
@@ -0,0 +1,82 @@
+use core::cell::UnsafeCell;
+use core::ops::Range;
+use core::ptr;
+use std::sync::Arc;
+
+#[derive(Clone, Debug)]
+pub struct Buffer {
+    /// This data is potentially accessed mutably in arbitrary non-overlapping slices,
+    /// so we must store it in `UnsafeCell` to avoid making any too-strong no-aliasing claims.
+    storage: Arc<UnsafeCell<[u8]>>,
+
+    /// Size of the allocation.
+    ///
+    /// This is redundant with `storage.get().len()`, but that method is not
+    /// available until our MSRV is 1.79 or greater.
+    size: usize,
+}
+
+/// SAFETY:
+/// This shared mutable data will not be accessed in a way which causes data races;
+/// the obligation to do so is on the caller of the HAL API.
+/// For safe code, `wgpu-core` validation manages appropriate access.
+unsafe impl Send for Buffer {}
+unsafe impl Sync for Buffer {}
+
+impl Buffer {
+    pub(in crate::noop) fn new(desc: &crate::BufferDescriptor) -> Result<Self, crate::DeviceError> {
```

Comment:
```suggestion
    pub(super) fn new(desc: &crate::BufferDescriptor) -> Result<Self, crate::DeviceError> {
```

Convention here is just pub(super)

Reply:
IMO, using `super`, particularly across files, is fragile and best kept to a minimum, because it means that module-moving refactors can have surprising effects. I’ll change it anyway, but I am noting that I disagree with this policy.

Reply:
Yeah that makes sense - I don't really have much of a preference either way, just following existing convention. Might be worth an issue about changing it. While hal backends have historically been a single mod + submodules, as we split things up more, we might see deeper splits.

Reply:
[Done.](https://github.com/gfx-rs/wgpu/issues/7121)